lib/web_ui/lib/src/engine/compositor/image.dart

```diff
@@ -72,7 +88,23 @@ class CkImage implements ui.Image {
   @override
   Future<ByteData> toByteData(
       {ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
-    throw 'unimplemented';
+    Uint8List bytes;
+
+    if (format == ui.ImageByteFormat.rawRgba) {
+      final SkImageInfo imageInfo = SkImageInfo(
+        alphaType: canvasKit.AlphaType.Premul,
+        colorType: canvasKit.ColorType.RGBA_8888,
+        width: width,
+        height: height,
+      );
+      bytes = skImage.readPixels(imageInfo, 0, 0);
+    } else {
+      final SkData skData = skImage.encodeToData(); //defaults to PNG 100%
+      bytes = canvasKit.getSkDataBytes(skData);
+    }
+
+    final ByteData data = Uint8List.fromList(bytes).buffer.asByteData(0, bytes.length);
```

Comment:
Same question here. If we need to copy the data, let's explain in a comment why.